ethcore/src/state/mod.rs

```diff
@@ -539,6 +539,13 @@ impl<B: Backend> State<B> {
 			|a| a.as_ref().map_or(self.account_start_nonce, |account| *account.nonce()))
 	}
 
+	/// Whether the base storage root of an account remains unchanged.
+	pub fn is_base_storage_root_unchanged(&self, a: &Address) -> TrieResult<bool> {
```

Comment:
why not just return `bool` here?

Reply:
Below `State::ensure_cached` can fail, and in that case I think it's better to bail because it's a DB failure.